File: 206-ReverseLinkedList/206-ReverseLinkedList.py
# Last updated: 7/2/2025, 11:53:22 PM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:

        dummy, temp, nextt = None, head, head

        while head:
            nextt = head.next
            head.next = dummy
            dummy = head
            head = nextt
        
        return dummy

        

File: 206-ReverseLinkedList/206-ReverseLinkedList.cpp
// Last updated: 7/2/2025, 11:53:18 PM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* dummy = NULL;
        //dummy->next = head;

        while(head){
            ListNode* next = head->next;
            head->next = dummy;
            dummy = head;
            head = next;
        }
        return dummy;
    }
};